src/saltext/vmware/states/esxi.py

```diff
@@ -905,56 +905,68 @@ def advanced_config(
         Use this vCenter service connection instance instead of creating a new one. (optional).
 
+    profile
+        Profile to use (optional)
+
+    less
+        Default False. If this is set to True, only the changed values will be reported as changes.
+
     .. code-block:: yaml
 
-        Remove User:
+        ESXi_advanced_config_example:
           vmware_esxi.advanced_configs:
-            - name: Annotations.WelcomeMessage
-            - value: Hello
-
+            - less: True
+            - configs:
+              key_1: value_1
+              key_2: value_2
     """
-    log.debug("Running vmware_esxi.advanced_config")
-    ret = {"name": name, "result": None, "comment": "", "changes": {}}
+    log.debug("Running vmware_esxi.advanced_configs")
     service_instance = service_instance or connect.get_service_instance(
         config=__opts__, profile=profile
     )
 
     esxi_config_old = __salt__["vmware_esxi.get_advanced_config"](
-        config_name=name,
+        config_name="",
         datacenter_name=datacenter_name,
         cluster_name=cluster_name,
         host_name=host_name,
         service_instance=service_instance,
     )
+
+    changes = {} if less else {"new": {}, "old": {}}
+
     if __opts__["test"]:
-        ret["result"] = None
-        ret["changes"] = {"new": {}}
         for host in esxi_config_old:
-            ret["changes"]["new"][host] = f"{name} will be set to {value}"
-        ret["comment"] = "These options are set to change."
-        return ret
+            diff = salt.utils.data.recursive_diff(esxi_config_old[host], configs)
+            if less:
+                changes[host] = {k: diff["old"][k] for k in diff["new"]}
+            else:
+                changes["new"][host] = [f"{k} will be set to {diff['new'][k]}" for k in diff["new"]]
+                changes["old"][host] = [f"{k} was {diff['old'][k]}" for k in diff["new"]]
+        return {"name": name, "result": True, "comment": "", "changes": changes}
 
-    ret["result"] = True
-    ret["changes"] = {"new": {}, "old": {}}
-    change = False
     for host in esxi_config_old:
-        if esxi_config_old[host][name] != value:
-            change = True
-            config = __salt__["vmware_esxi.set_advanced_configs"](
-                config_dict={name: value},
-                datacenter_name=datacenter_name,
-                cluster_name=cluster_name,
-                host_name=host,
-                service_instance=service_instance,
-            )
-            ret["changes"]["old"][host] = f"{name} was {esxi_config_old[host][name]}"
-            ret["changes"]["new"][host] = f"{name} was changed to {config[host][name]}"
-
-    if change:
-        ret["comment"] = "Configurations have successfully been changed."
-    else:
-        ret["comment"] = "Configurations are already in correct state."
-    return ret
+        for name in configs:
+            value = configs[name]
+            if esxi_config_old[host][name] != value:
```

Comment:
**suggestion** if the config key is not present in the old key, this will error. I'm not sure what the correct/expected behavior should be. We can either create an issue to track this fix, or fix it as part of this PR. I used the literal example (key_1: value_1) when testing this, and that's when it blew up for me :sweat_smile:

Reply:
I updated the example with real values and handled the case when the key is not present on any host.